ads/google/a4a/utils.js

```diff
@@ -357,92 +356,71 @@ export function additionalDimensions(win, viewportSize) {
 /**
  * Extracts configuration used to build amp-analytics element for active view.
  *
+ * @param {!../../../extensions/amp-a4a/0.1/amp-a4a.AmpA4A} a4a
  * @param {!../../../src/service/xhr-impl.FetchResponseHeaders} responseHeaders
  *   XHR service FetchResponseHeaders object containing the response
  *   headers.
- * @param {!../../../src/service/extensions-impl.Extensions} extensions
- * @return {?AmpAnalyticsConfigDef} config or null if invalid/missing.
+ * @return {?JSONType} config or null if invalid/missing.
  */
-export function extractAmpAnalyticsConfig(responseHeaders, extensions) {
+export function extractAmpAnalyticsConfig(a4a, responseHeaders) {
   if (responseHeaders.has(AMP_ANALYTICS_HEADER)) {
     try {
       const analyticsConfig =
         JSON.parse(responseHeaders.get(AMP_ANALYTICS_HEADER));
-      dev().assert(Array.isArray(analyticsConfig['url']));
-      if (analyticsConfig.url.length) {
-        extensions.loadExtension('amp-analytics');
```

Comment:
I explicitly want to load the extension when parsing the header once we verify its validity.  This is because it occurs during onLayoutMeasure trigger ad request promise chain while actual writing of the amp-analytic element occurs during layoutCallback.  Given there could be a delay between these events, I want to ensure that amp analytics exists and is loaded as early as possible in order to reduce any latency to the visibility ping.